hack/lib/cleanup.sh

```diff
@@ -16,18 +16,25 @@ function os::cleanup::dump_etcd() {
 	os::util::curl_etcd "/v2/keys/?recursive=true" > "${ARTIFACT_DIR}/etcd_dump.json"
 }
 
-# kill_all_processes function will kill all
-# all processes created by the test script.
-function kill_all_processes() {
+# os::cleanup::jobs cleans up running jobs spawned by this script.
+# We assume that the jobs are well-formed to respond to SIGTERM and
+# that they correctly forward the signal to their children.
+#
+# Globals:
+#  - USE_SUDO
+# Arguments:
+#  None
+# Returns:
+#  None
+function os::cleanup::jobs() {
 	local sudo="${USE_SUDO:+sudo}"
 
-	pids=($(jobs -pr))
-	for i in "${pids[@]-}"; do
-		pgrep -P "${i}" | xargs ${sudo} kill &> /dev/null
```

Comment:
We should just assume a process kills it's children, or go down the whole tree recursively